### PR TITLE
Space extension first pass

### PR DESCRIPTION
**Pull request by srietkerk**

### CODE REVIEW

File: icon.png
Comment (jwunderl) [inline icon.png:1]: This file should be shrunk to ~100-500kb

File: planets.ts [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+        Saturn = 185,
+        Uranus = 358,
+        Neptune = 562,
+    }
+
+    /**
+     * Creates a solar system.
+     * @param scale scale of radius of planets; eg: 1, 5; Multiplies the actual radius by (scale/3)
+     */
+    //% blockId=space_solarsystem block="solar system at center $center=minecraftCreatePosition || of scale $scale"
```
Comment (riknoll) [inline planets.ts]:
```suggestion
    //% blockId=space_solarsystem block="solar system at center $center|| of scale $scale"
    //% center.shadow=minecraftCreatePosition
```

File: planets.ts [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+        createJupiter(jupiterCenter, Radius.Jupiter * scale);
+        player.teleport(saturnCenter.add(pos(0, 0, Radius.Saturn + 5)));
+        createSaturn(saturnCenter, Radius.Saturn * scale);
+        player.teleport(uranusCenter.add(pos(0, 0, Radius.Uranus + 5)));
+        createUranus(uranusCenter, Radius.Uranus * scale);
+        player.teleport(neptuneCenter.add(pos(0, 0, Radius.Neptune + 5)));
+        createNeptune(neptuneCenter, Radius.Neptune * scale);
+    }
+
+    //% blockId=space_planet_mercury block="mercury at center $center=minecraftCreatePosition || of radius $radius"
```
Comment (riknoll) [inline planets.ts]:
```suggestion
    //% blockId=space_planet_mercury block="mercury at center $center|| of radius $radius"
    //% center.shadow=minecraftCreatePosition
```

File: planets.ts [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        if (radius <= 0) return;
+        space.sphere(
+            LIGHT_GRAY_CONCRETE,
+            center.add(pos(0, radius, 0)),
+            radius,
+            ShapeOperation.Outline
+        );
+    }
+
+    //% blockId=space_planet_venus block="venus at center $center=minecraftCreatePosition || of radius $radius"
```
Comment (riknoll) [inline planets.ts]:
```suggestion
    //% blockId=space_planet_venus block="venus at center $center|| of radius $radius"
    //% center.shadow=minecraftCreatePosition
```

File: planets.ts [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+    //% weight=98
+    export function createVenus(
+        center: Position,
+        radius?: number
+    ): void {
+        if (radius <= 0) return;
+        space.sphere(YELLOW_CONCRETE, center.add(pos(0, radius, 0)), radius, ShapeOperation.Outline);
+    }
+
+    //% blockId=space_planet_earth block="earth at center $center=minecraftCreatePosition || of radius $radius"
```
Comment (riknoll) [inline planets.ts]:
```suggestion
    //% blockId=space_planet_earth block="earth at center $center|| of radius $radius"
    //% center.shadow=minecraftCreatePosition
```

File: planets.ts [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+            [LIGHT_BLUE_CONCRETE, GRASS, WHITE_CONCRETE],
+            center.add(pos(0, radius, 0)),
+            radius,
+            Axis.Y,
+            true,
+            ShapeOperation.Outline
+        );
+    }
+
+    //% blockId=space_planet_mars block="mars at center $center=minecraftCreatePosition || of radius $radius"
```
Comment (riknoll) [inline planets.ts]:
```suggestion
    //% blockId=space_planet_mars block="mars at center $center|| of radius $radius"
    //% center.shadow=minecraftCreatePosition
```

File: planets.ts [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+    //% weight=96
+    export function createMars(
+        center: Position,
+        radius?: number
+    ): void {
+        if (radius <= 0) return;
+        space.sphere(RED_SANDSTONE, center.add(pos(0, radius, 0)), radius, ShapeOperation.Outline);
+    }
+
+    //% blockId=space_planet_jupiter block="jupiter at center $center=minecraftCreatePosition || of radius $radius"
```
Comment (riknoll) [inline planets.ts]:
```suggestion
    //% blockId=space_planet_jupiter block="jupiter at center $center|| of radius $radius"
    //% center.shadow=minecraftCreatePosition
```

File: planets.ts [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+            [ORANGE_CONCRETE, WHITE_CONCRETE],
+            center.add(pos(0, radius, 0)),
+            radius,
+            Axis.Y,
+            true,
+            ShapeOperation.Outline
+        );
+    }
+
+    //% blockId=space_planet_saturn block="saturn at center $center=minecraftCreatePosition || of radius $radius"
```
Comment (riknoll) [inline planets.ts]:
```suggestion
    //% blockId=space_planet_saturn block="saturn at center $center|| of radius $radius"
    //% center.shadow=minecraftCreatePosition
```

File: planets.ts [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+                center.add(pos(0, radius, 0)),
+                radius + 3 + i * 2,
+                Axis.Y,
+                ShapeOperation.Outline
+            );
+        }
+
+    }
+
+    //% blockId=space_planet_uranus block="uranus at center $center=minecraftCreatePosition || of radius $radius"
```
Comment (riknoll) [inline planets.ts]:
```suggestion
    //% blockId=space_planet_uranus block="uranus at center $center|| of radius $radius"
    //% center.shadow=minecraftCreatePosition
```

File: planets.ts [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+        if (radius <= 0) return;
+        space.sphere(
+            LIGHT_BLUE_CONCRETE,
+            center.add(pos(0, radius, 0)),
+            radius,
+            ShapeOperation.Outline
+        );
+    }
+
+    //% blockId=space_planet_neptune block="neptune at center $center=minecraftCreatePosition || of radius $radius"
```
Comment (riknoll) [inline planets.ts]:
```suggestion
    //% blockId=space_planet_neptune block="neptune at center $center|| of radius $radius"
    //% center.shadow=minecraftCreatePosition
```

File: planets.ts [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+    //% weight=92
+    export function createNeptune(
+        center: Position,
+        radius?: number
+    ): void {
+        if (radius <= 0) return;
+        space.sphere(BLUE_CONCRETE, center.add(pos(0, radius, 0)), radius, ShapeOperation.Outline);
+    }
+
+    //% blockId=space_stars_sun block="sun at center $center=minecraftCreatePosition || of radius $radius"
```
Comment (riknoll) [inline planets.ts]:
```suggestion
    //% blockId=space_stars_sun block="sun at center $center|| of radius $radius"
    //% center.shadow=minecraftCreatePosition
```

File: sphere.ts [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+            }
+        }
+    }
+
+    /**
+     * Fill a sphere of blocks chosen at random from the block list provided, at a center position.
+     * @param radius the radius of the sphere, eg: 5
+     */
+    //% blockId=space_sphereRandomBlocks
+    //% block="sphere of random blocks from $blockList|center $center=minecraftCreatePosition|radius $radius|$operator"
```
Comment (riknoll) [inline sphere.ts]:
```suggestion
    //% block="sphere of random blocks from $blockList|center $center|radius $radius|$operator"
    //% center.shadow=minecraftCreatePosition
```

File: sphere.ts [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+            }
+        }
+    }
+
+    /**
+     * Fill a sphere of blocks where block for each layer is chosen at random or in order from the block list provided, at a center position.
+     * @param radius the radius of the sphere, eg: 5
+     */
+    //% blockId=space_sphereRandomLayerBlocks
+    //% block="sphere of random layers from $blockList|center $center=minecraftCreatePosition|radius $radius|around $orientation|in order of blocks $useBlockOrder|$operator"
```
Comment (riknoll) [inline sphere.ts]:
```suggestion
    //% block="sphere of random layers from $blockList|center $center|radius $radius|around $orientation|in order of blocks $useBlockOrder|$operator"
    //% center.shadow=minecraftCreatePosition
```

File: README.md [hunk 13]
```diff
@@ -1,33 +1,14 @@
-# Project
+## MakeCode for Minecraft Education Extension
 
-> This repo has been populated by an initial template to help get you started. Please
-> make sure to update the content to build a great experience for community-building.
+This repository is an approved **extension** in Makecode for Minecraft Education.
```
Comment (riknoll) [inline README.md]:
```suggestion
This repository is an approved **extension** in Microsoft MakeCode for Minecraft Education. 
```

**Review by riknoll — CHANGES_REQUESTED** [review]
Restore the contributing and trademark sections in the README!

Also, doesn't have to be this PR but you need to add docs and JSDoc comments for all of the functions

**Review by riknoll — APPROVED** [review]

**Pull request merged**